cardano-cli/src/Cardano/CLI/EraBased/Run/CreateTestnetData.hs

```diff
@@ -227,20 +227,20 @@ runGenesisCreateTestNetDataCmd Cmd.GenesisCreateTestNetDataCmdArgs
 
   when (0 < numUtxoKeys) $ writeREADME utxoKeysDir utxoKeysREADME
 
-  mayStakePoolRelays
-    <- forM relays $
-       \fp -> do
-         relaySpecJsonBs <-
-           handleIOExceptT (GenesisCmdStakePoolRelayFileError fp) (LBS.readFile fp)
-         firstExceptT (GenesisCmdStakePoolRelayJsonDecodeError fp)
-           . hoistEither $ Aeson.eitherDecode relaySpecJsonBs
+  mSPOsRelays <- forM relays readRelays
+  case (relays, mSPOsRelays) of
+    (Just fp, Just stakePoolRelays) | Map.size stakePoolRelays > fromIntegral numPools ->
+      throwError $ GenesisCmdTooManyRelaysError fp (fromIntegral numPools) (Map.size stakePoolRelays)
+    _ -> pure ()
```

Comment:
We can avoid the wildcard pattern (IMO this is less readable) match if you case on `relays` i.e
```
case relays of 
  Nothing -> pure ()
  Just relaySpecFile -> ...
```

Reply:
@Jimbo4350> not much shorter with your version, because I need mSPOsRelays to be a `Just` to look at the size of the underlying Map. If I don't pattern match on it, I need to default `Nothing` to being of size `0`, which is a bit cumbersome.

So I'll skip this one.